File: LeetCode Solutions/March 2022 LeetCoding Challenges/Arithmetic Slices/Arithmetic Slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums) {
        int arithmeticSlices = 0, combination = 0, lastDifference = 0;
        for (int i=0; i<nums.size()-1; i++){
            int difference = nums[i+1] - nums[i];
            if (i != 0 && difference == lastDifference){
                combination += 1;
                arithmeticSlices += combination;
            } else{
                combination = 0;
            }
            lastDifference = difference;
        }
        return arithmeticSlices;
    }
};

File: LeetCode Solutions/March 2022 LeetCoding Challenges/Arithmetic Slices/Arithmetic Slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        arithmeticSlices = combination = lastDifference = 0
        for i in range(len(nums)-1):
            difference = nums[i+1] - nums[i]
            if (i != 0 and difference == lastDifference):
                combination += 1
                arithmeticSlices += combination
            else:
                combination = 0
            lastDifference = difference
        return arithmeticSlices